src/main/java/com/capitalone/dashboard/collector/SonarCollectorTask.java

```diff
@@ -248,69 +248,86 @@ private void deleteUnwantedJobs(List<SonarProject> latestProjects, List<SonarPro
     private void refreshData(List<SonarProject> sonarProjects, SonarClient sonarClient) {
         long start = System.currentTimeMillis();
         int count = 0;
-
+        int updated = 0;
+        int disabled = 0;
         for (SonarProject project : sonarProjects) {
-            CodeQuality codeQuality = sonarClient.currentCodeQuality(project);
-            if (codeQuality != null && isNewQualityData(project, codeQuality)) {
-                project.setLastUpdated(System.currentTimeMillis());
-                sonarProjectRepository.save(project);
-                codeQuality.setCollectorItemId(project.getId());
-                codeQualityRepository.save(codeQuality);
-                count++;
+            try {
+                CodeQuality codeQuality = sonarClient.currentCodeQuality(project);
+                if (codeQuality != null && isNewQualityData(project, codeQuality)) {
+                    project.setLastUpdated(System.currentTimeMillis());
+                    sonarProjectRepository.save(project);
+                    codeQuality.setCollectorItemId(project.getId());
+                    codeQualityRepository.save(codeQuality);
+                    updated++;
+                }
+            } catch (HttpClientErrorException e) {
+                String body = e.getResponseBodyAsString();
+                if ((e.getStatusCode() == HttpStatus.NOT_FOUND) && body != null && body.contains("errors") && body.contains("not found")) {
+                    project.setEnabled(false);
+                    sonarProjectRepository.save(project);
+                    LOG.info("Disabled as a result of HTTPStatus.NOT_FOUND, projectName=" + project.getProjectName()
```

Comment:
@Courtneyp123  this line can throw NPE because there is chance when projectId is null and if 404 occurred due to projectId being null, that will be NPE.